web/src/routes/SearchPage.tsx

```diff
@@ -82,8 +90,6 @@ const SearchPage = ({ city, cityCode, languageCode, pathname }: CityRouteProps):
 
   const handleFilterTextChanged = (filterText: string): void => {
     setFilterText(filterText)
-    const appendToUrl = filterText.length !== 0 ? `?query=${filterText}` : ''
-    navigate(`${pathname}/${appendToUrl}`, { replace: true })
```

Comment:
:wrench: As mentioned above, I think we should directly update the url, so this should probably be reverted. With the change from above, you could then do the following here:
``` ts
    setSearchParams({ query })
    const appendToUrl = query.length !== 0 ? `?query=${query}` : ''
    navigate(`${pathname}/${appendToUrl}`, { replace: true })
```